packages/flutter/lib/src/material/range_slider.dart

```diff
@@ -1513,37 +1514,102 @@ class _RenderRangeSlider extends RenderBox with RelayoutWhenSystemFontsChangeMix
 
-    config.isSemanticBoundary = isEnabled;
+  // Create the semantics configuration for a single value.
+  SemanticsConfiguration _createSemanticsConfiguration(
+      double value,
+      String label,
+      VoidCallback increaseAction,
+      VoidCallback decreaseAction,
+  ) {
+    final SemanticsConfiguration config = SemanticsConfiguration();
+    config.isEnabled = isEnabled;
+    config.textDirection = textDirection;
     if (isEnabled) {
-      config.textDirection = textDirection;
-      config.customSemanticsActions = <CustomSemanticsAction, VoidCallback>{
-        _decreaseStart: _decreaseStartAction,
-        _increaseStart: _increaseStartAction,
-        _decreaseEnd: _decreaseEndAction,
-        _increaseEnd: _increaseEndAction,
-      };
-      if (semanticFormatterCallback != null) {
-        config.value = semanticFormatterCallback(_state._lerpRangeValues(values));
-      } else {
-        config.value = values.toString();
-      }
+      config.onIncrease = increaseAction;
+      config.onDecrease = decreaseAction;
     }
+    config.label = label ?? '';
+    if (semanticFormatterCallback != null) {
+      config.value = semanticFormatterCallback(_state._lerp(value));
+      config.increasedValue = semanticFormatterCallback(_state._lerp((value + _semanticActionUnit).clamp(0.0, 1.0) as double));
+      config.decreasedValue = semanticFormatterCallback(_state._lerp((value - _semanticActionUnit).clamp(0.0, 1.0) as double));
+    } else {
+      config.value = '${(value * 100).round()}%';
+      config.increasedValue = '${((value + _semanticActionUnit).clamp(0.0, 1.0) * 100).round()}%';
+      config.decreasedValue = '${((value - _semanticActionUnit).clamp(0.0, 1.0) * 100).round()}%';
+    }
+
+    return config;
   }
 
-  final CustomSemanticsAction _decreaseStart = const CustomSemanticsAction(label: 'Decrease Min');
-  final CustomSemanticsAction _increaseStart = const CustomSemanticsAction(label: 'Increase Min');
-  final CustomSemanticsAction _decreaseEnd = const CustomSemanticsAction(label: 'Decrease Max');
-  final CustomSemanticsAction _increaseEnd = const CustomSemanticsAction(label: 'Increase Max');
+  @override
+  void assembleSemanticsNode(
+      SemanticsNode node,
+      SemanticsConfiguration config,
+      Iterable<SemanticsNode> children,
+  ) {
+    final SemanticsConfiguration startSemanticsConfiguration = _createSemanticsConfiguration(
+        values.start,
+        labels?.start,
+        _increaseStartAction,
+        _decreaseStartAction,
+    );
+    final SemanticsConfiguration endSemanticsConfiguration = _createSemanticsConfiguration(
+        values.end,
+        labels?.end,
+        _increaseEndAction,
+        _decreaseEndAction,
+    );
+
+    // Split the semantics node area between the start and end nodes.
+    final Rect leftRect = Rect.fromPoints(node.rect.topLeft, node.rect.bottomCenter);
+    final Rect rightRect = Rect.fromPoints(node.rect.topCenter, node.rect.bottomRight);
```

Comment:
This means, the accessibility focus rect (that green rect that Talkback draws) will always cover the first half of the slider for the first thump and the second half of the slider for the second thump - no matter the thumb position, right? Is that the expected behavior for a range slider? I would have expected the semantics nodes to be around the thumbs...

Reply:
Correct, the thinking was that splitting the area in two makes the widget more accessible due to the larger targets (the slider widget's semantic node is around the whole slider, for instance). It also helps when the two sliders are on top of one another